CenterForOpenScience/osf.io
# [Py3][ENG-491] Ultimate backwards compatible Python 3

## Purpose

Make all possible backward compatible Python 3 changes.

## Changes
- minor dependency updates for addons.
- list wrap all dictviews
- stringify exception messages `e.message` -> `str(e)`
- change cmp argument to key for `sorted` 
- change asserts statement for new comparisons 
- remove unnecessary auth statements that were causing environmental errors
- remove unsed decorator `http_error_if_disk_saving_mode`
- changes ExternalProvider metaclass syntax to be both py2/3 compatible 

## QA Notes
- check order for filtering
- check that renaming publicly wiki
- Elastic search
- ownclould/twofactor have version bump

## Documentation

Not user facing

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-491
